apps/wallet-mobile/src/features/AddWallet/common/StepperProgress/StepperProgress.tsx

```diff
@@ -0,0 +1,118 @@
+import {useTheme} from '@yoroi/theme'
+import React from 'react'
+import {StyleSheet} from 'react-native'
+import Animated, {Layout} from 'react-native-reanimated'
+
+import CheckIcon from '../../illustrations/CheckIcon'
+
+type StepProps = {
+  currentStep: number
+  currentStepTitle?: string
+  todoStep: boolean
+  pastStep: boolean
+  displayStepNumber?: boolean
+}
+const Step = ({currentStep, currentStepTitle, displayStepNumber, todoStep, pastStep}: StepProps) => {
+  const {styles} = useStyles()
+  return (
+    <>
+      <Animated.View
+        layout={Layout}
+        style={[
+          styles.step,
+          todoStep && styles.todoStep,
+          pastStep && styles.pastStep,
+          displayStepNumber === true && styles.markedStep,
+        ]}
+      >
+        {displayStepNumber === true && !pastStep ? (
+          <Animated.Text style={[styles.stepNumber, todoStep && styles.todoStepNumber]}>{currentStep}</Animated.Text>
+        ) : (
+          <CheckIcon />
+        )}
+      </Animated.View>
+
+      {!todoStep && !pastStep && currentStepTitle !== undefined && (
```

Comment:
build a var, that identifies this logic, what means !todoStep && !pastStep ... yada yada?
```
const shouldDisplayCurrentStep = ... bla bla bla
```